keras/engine/training.py

```diff
@@ -1311,6 +1311,11 @@ def predict(self, x,
 
         Computation is done in batches.
 
+        Note: the internal prediction function is not compiled until the first
+        call to `Model.predict`. If you are using `Model.predict` in a
+        multi-threaded environment, you should explicitly call
+        `Model._make_predict_function` first to avoid problems.
```

Comment:
`_make_predict_function` is a private API. We should not recommend calling it.

Here, the user should simply call `predict` first.

Note that Keras models can't be guaranteed to be thread-safe. Consider having independent copies of the model in each thread for CPU inference.